ibm/service/database/validators.go

```diff
@@ -0,0 +1,146 @@
+// Copyright IBM Corp. 2017, 2022 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package database
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/IBM/cloud-databases-go-sdk/clouddatabasesv5"
+)
+
+/* TODO move other validators in here */
+
+/* VERSION VALIDATOR */
+
+type Version struct {
+	Version     string
+	Type        string
+	Status      string
+	IsPreferred bool
+	Transitions []VersionTransition
+}
+
+type VersionTransition struct {
+	Application         string
+	Method              string
+	SkipBackupSupported bool
+	FromVersion         string
+	ToVersion           string
+}
+type AllowedUpgrade struct {
+	ToVersion           string `json:"to_version"`
+	SkipBackupSupported bool   `json:"skip_backup_supported"`
+}
+
+func getAllowedUpgradeVersions(versions []Version) []AllowedUpgrade {
```

Comment:
This can be a method on the Version struct i.e. 
```
func (v *Version) CanUpgradeTo(newVersion string) bool {
  // ...
  return false
}
```

Reply:
@alexhemard I did that for the `isVersionAllowed` function here https://github.com/lornakelly/terraform-provider-ibm/pull/2/files#diff-becda8b3a549e47ecef37bc08a1c7400774f65d3e6fb3fb4cf85bab55522964fR54

`getAllowedUpgradeVersions` returns an array of allowed versions not a bool. Should I still try add as a method on Version struct?